### PR TITLE
Remove obsolete form factor parametrizations

### PR DESCRIPTION
**Pull request by dvandyk**
Removed:
 - ``BZ2004`` and ``BZ2004v2``
 - ``BFW2010`` (only the simplified series expansion was implemented)
 - ``FMvD2015``

In furtherance of #499.

### CODE REVIEW

File: eos/form-factors/mesonic.cc [hunk 1]
```diff
@@ -3,7 +3,6 @@
 /*
  * Copyright (c) 2010, 2011, 2013, 2014, 2015, 2016, 2018 Danny van Dyk
  * Copyright (c) 2015 Christoph Bobeth
- * Copyright (c) 2010, 2011 Christian Wacker
```
Comment (mreboud) [inline eos/form-factors/mesonic.cc:6]: Why did you removed this entry?

**Review by mreboud — APPROVED** [review]
Github complains about missing newlines at the end of most of the new files.
I don't have comments, thanks for having taken care of this!

**Comment by mreboud** [comment]
The parametrizations still need to be removed from:

- [ ] The parameters files
- [ ] All the test case (@mreboud 's job)
- For the doc it should be automatic...

**Comment by mreboud** [comment]
Pff for `FMvD2015`, it is also used in the test case. I will trade it for `BSZ2015`, but the default values for `B_s -> K^*` are not in EOS... So I will start by adding the values for the original `BSZ` paper, and then update all the test cases

File: eos/form-factors/mesonic.cc (hunk 1 above)
Comment (dvandyk) [reply eos/form-factors/mesonic.cc:6]: Because his code was moved to a different file.

Comment (mreboud) [reply eos/form-factors/mesonic.cc:6]: Ah I see, good

**Comment by dvandyk** [comment]
> Pff `FMvD2015` is also used in the test case. I will trade it for `BSZ2015`, but the default values for `B_s -> K^*` are not in EOS... So I will start by adding the values of the original `BSZ` paper, and then update all the test cases

Thank you!

**Pull request merged**